web/client/src/library/components/graph/help.ts

```diff
@@ -474,34 +444,295 @@ function mergeConnections(
+  withSecondary: boolean = false,
+): Edge[] {
+  const tempEdges = edges.map(edge => {
+    const isActiveEdge = hasActiveEdge(activeEdges, [
+      edge.targetHandle,
+      edge.sourceHandle,
+    ])
+
+    edge.hidden = true
+
+    if (isNil(edge.sourceHandle) && isNil(edge.targetHandle)) {
+      // Edge between models
+      const hasSelections =
+        selectedNodes.size > 0 || connections.size > 0 || activeNodes.size > 0
+      const isImpactEdge =
+        connectedNodes.has(edge.source) || connectedNodes.has(edge.target)
+      const isSecondaryEdge =
+        isFalse(connectedNodes.has(edge.source)) ||
+        isFalse(connectedNodes.has(edge.target))
+      const withoutImactNodes =
+        isFalse(withImpact) && isFalse(withConnected) && isFalse(hasSelections)
+      const withoutSecondaryNodes =
+        isFalse(withSecondary) && isFalse(hasSelections)
+      const shouldHideSecondary = isSecondaryEdge && withoutSecondaryNodes
+      const shouldHideImpact = isImpactEdge && withoutImactNodes
+      const isVisibleEdge =
+        selectedNodes.size > 0 &&
+        selectedEdges.has(edge.id) &&
+        activeNodes.has(edge.source) &&
+        activeNodes.has(edge.target)
+
+      if (
+        isFalse(shouldHideImpact) &&
+        isFalse(shouldHideSecondary) &&
+        (isFalse(hasSelections) || isVisibleEdge)
+      ) {
+        edge.hidden = false
+      }
+    } else {
+      // Edge between columns
+      if (connections.size > 0 && isActiveEdge) {
+        edge.hidden = false
+      }
+    }
+
+    let stroke = 'var(--color-graph-edge-main)'
+    let strokeWidth = 2
+
+    const isConnectedSource = connectedNodes.has(edge.source)
+    const isConnectedTarget = connectedNodes.has(edge.target)
+
+    if (
+      selectedEdges.has(edge.id) ||
+      (withConnected && isConnectedSource && isConnectedTarget)
+    ) {
+      strokeWidth = 4
+      stroke = 'var(--color-graph-edge-selected)'
+      edge.zIndex = 10
+    } else {
+      if (isActiveEdge) {
+        stroke = 'var(--color-graph-edge-secondary)'
+      } else if (isConnectedSource && isConnectedTarget) {
+        strokeWidth = 4
+        stroke = 'var(--color-graph-edge-direct)'
+        edge.zIndex = 10
+      }
+    }
+
+    edge.style = {
+      ...edge.style,
+      stroke,
+      strokeWidth,
+    }
+
+    return edge
+  })
+
+  return tempEdges
+}
+
+function getUpdatedNodes(
+  nodes: Node[] = [],
+  activeNodes: Set<string>,
+  mainNode: string,
+  connectedNodes: Set<string>,
+  selectedNodes: Set<string>,
+  connections: Map<string, Connections>,
+  withConnected: boolean = true,
+  withImpact: boolean = true,
+  withSecondary: boolean = true,
+  withColumns: boolean = true,
+): Node[] {
+  return nodes.map(node => {
+    node.hidden = true
+
+    const hasSelections = selectedNodes.size > 0 || connections.size > 0
+    const isActiveNode = activeNodes.size === 0 || activeNodes.has(node.id)
+    const isImpactNode = connectedNodes.has(node.id)
```

Comment:
What is an impact node?

Reply:
represent a `model` that will be impacted by change in `current model` or  a `model` that impacts `current model`

Reply:
Thanks for clarifying. Nit: what do you think about renaming this to `isImpactedNode`?